File: src/acmicpc/2019/old/11047.py
import sys
q = lambda: sys.stdin.readline().strip()
n,k = list(map(int,q().split(' ')))
coins = []
for i in range(n):
	coins.append(int(q()))
coins = list(reversed(coins))

coin = 0
for i in coins:
	if k // i != 0:
		coin += k//i
		k = k%i
print(coin)

File: src/acmicpc/2019/old/11047.cpp
#include <iostream>
using namespace std;
int main(void){
	int n,k,a[11],coin=0;
	
	cin>>n>>k;
	for(int i=1;i<=n;i++)
		cin>>a[i];
	while(k>0){
		if(a[n]<=k){
			coin += k/a[n];
			k = k%a[n];
		}
		n--;
	}
	cout<<coin;
}